### PR TITLE
Remove sprockets

### PR DESCRIPTION
**Pull request by JamesGlover**
This builds on the removal of webpacker https://github.com/sanger/sequencescape/pull/3536 in attempts to unify our entire asset pipeline.

Things get a bit more complicated here, as there is a lot more to move, and some of it is pretty ancient.

I actually found a fair bit of broken JS while doing this, and either fixed it up (where it appeared useful) or dropped it where it was superfluous.

Still some things to fix up here.

### CODE REVIEW

**Comment by JamesGlover** [comment]
I've made this ready for review. It still uses sprockets for Images and CSS, but gets rid of its use for js.

**Comment by JamesGlover** [comment]
And back to draft, as I think I can do some useful cherrypicking to break this one down. (Ironically to remove flexible cherrypicking)

File: app/frontend/entrypoints/organism_validation.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/*====================================================================
```
Comment (JamesGlover) [inline app/frontend/entrypoints/organism_validation.js:1]: Actually migrated from:
app/assets/javascripts/organism_validation.js

File: app/frontend/entrypoints/organism_validation.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ *  Author: Tony Cox (avc@sanger.ac.uk)
```
Comment (stevieing) [inline app/frontend/entrypoints/organism_validation.js:2]: Tony Cox!!!

File: config/application.rb [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+require 'active_storage/engine'
+require 'action_controller/railtie'
+require 'action_view/railtie'
+require 'action_mailer/railtie'
+require 'active_job/railtie'
+require 'action_cable/engine'
+require 'action_mailbox/engine'
+require 'action_text/engine'
+require 'rails/test_unit/railtie'
+# require 'sprockets/railtie'
```
Comment (stevieing) [inline config/application.rb:18]: I guess this comment can be removed?

**Review by stevieing — APPROVED** [review]

File: config/application.rb (hunk 3 above)
Comment (JamesGlover) [reply config/application.rb:18]: I left it in there for reference ; it is part of the standard 'rails/all' inclusions and I wanted the two to be comparable to assist with upgrades. Sprockets is dropped in the default Rails 7 install though, so will definitely be able to remove it then.

We can probably also kill a few others of these, but that is probably for another PR.

**Comment by JamesGlover** [comment]
Cool, I'll see if I can get a test build and deploy into training or UAT on Tuesday, before I go ahead and merge this, just in case I break the build unexpectedly.

**Pull request merged**